mteb/tasks/Retrieval/eng/LitSearchRetrieval.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import annotations
+
+import datasets
+
+from mteb.abstasks.AbsTaskRetrieval import AbsTaskRetrieval
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+
+class LitSearchRetrieval(AbsTaskRetrieval):
+    metadata = TaskMetadata(
+        name="LitSearchRetrieval",
+        description="""
+        The dataset contains the query set and retrieval corpus for the paper LitSearch: A Retrieval Benchmark for
+        Scientific Literature Search. It introduces LitSearch, a retrieval benchmark comprising 597 realistic literature
+        search queries about recent ML and NLP papers. LitSearch is constructed using a combination of (1) questions
+        generated by GPT-4 based on paragraphs containing inline citations from research papers and (2) questions about
+        recently published papers, manually written by their authors. All LitSearch questions were manually examined or
+        edited by experts to ensure high quality.
+        """,
+        reference="https://github.com/princeton-nlp/LitSearch",
+        dataset={
+            "path": "princeton-nlp/LitSearch",
+            "revision": "9573fb284a1026c998df47024b888a163f0f0e25",
+        },
+        type="Retrieval",
+        category="s2p",
+        modalities=["text"],
+        eval_splits=["test"],
+        eval_langs=["eng-Latn"],
+        main_score="ndcg_at_10",
+        date=None,
+        domains=["Academic", "Non-fiction", "Written"],
+        task_subtypes=["Article retrieval"],
+        license="MIT",
+        annotations_creators="derived",
+        dialect=None,
+        sample_creation="found",
```

Comment:
Question are LLM generated, but samples are found. So I would probably say found is fine here. Though I would add a comment.

Reply:
Yup, you're right, the corpus comes from ACL Anthology and ICLR papers extracted from S2ORC